src/main/kotlin/graphql/kickstart/tools/SchemaParserOptions.kt

```diff
@@ -16,20 +16,22 @@ import org.reactivestreams.Publisher
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.CompletionStage
 import java.util.concurrent.Future
+import javax.validation.Validator
 import kotlin.coroutines.CoroutineContext
 import kotlin.reflect.KClass
 
 data class SchemaParserOptions internal constructor(
     val contextClass: Class<*>?,
     val genericWrappers: List<GenericWrapper>,
     val allowUnimplementedResolvers: Boolean,
-    val objectMapperProvider: PerFieldObjectMapperProvider,
+    val objectMapperProvider: PerFieldProvider<ObjectMapper>,
     val proxyHandlers: List<ProxyHandler>,
     val preferGraphQLResolver: Boolean,
     val introspectionEnabled: Boolean,
     val coroutineContextProvider: CoroutineContextProvider,
     val typeDefinitionFactories: List<TypeDefinitionFactory>,
-    val fieldVisibility: GraphqlFieldVisibility?
+    val fieldVisibility: GraphqlFieldVisibility?,
+    val validatorProvider: PerFieldProvider<Validator>?
```

Comment:
Kinda improvising here, not sure if it makes sense to have a public API that allows for customizing the validator instance per field level.